### PR TITLE
doc: Added enhancements to security doc

### PR DESCRIPTION
**Pull request by ashokdevtron**
**Main Changes**
- Added enhancements to application-level scanning (image scan, code scan, manifest scan)
- Connected helm chart scanning too
- Included gifs for better understanding

**Adhoc Changes**
- Fixed broken links in glossary and application groups
- Added a redirection from old cluster doc to resource browser doc

### CODE REVIEW

File: docs/user-guide/security-features.md [hunk 1]
```diff
@@ -1,20 +1,112 @@
...
-Install any one of the following integrations for scanning vulnerabilities:
-* [Clair](../user-guide/integrations/clair.md)
-* Trivy
-{% endhint %}
+Devtron provides [DevSecOps](https://devtron.ai/product/devsecops) capabilities across your software development life cycle for both: the default CI/CD solution by Devtron as well as your existing CI/CD Tools.
+
+One of the key components of DevSecOps is the detection of security risks. Currently, Devtron supports the following types of scanning:
+
+* Image Scan
+* Code Scan (not available for Helm apps)
```
Comment (abhibhaw) [inline docs/user-guide/security-features.md:10]: It's not like not available for helm apps, helm apps doesn't have codes attached with the chart to be scanned. This seems like we haven't supported it

File: docs/user-guide/security-features.md [hunk 2]
```diff
@@ -1,20 +1,112 @@
...
+* Vulnerability
+* License Risks
+* Misconfigurations
+* Exposed Secrets
+
+---
+
+## Where to Initiate the Scan
+
+### After Code Commit
```
Comment (abhibhaw) [inline docs/user-guide/security-features.md:26]: This doesn't guarantee a scan post code commit, it does before building the image

File: docs/user-guide/security-features.md [hunk 3]
```diff
@@ -1,20 +1,112 @@
...
+
+Results of Pre-CD scan will be visible under `Image Scan` and `Kubernetes Manifest` in the **App Details** page as shown below.
+
+![Figure 8: Pre-CD Scan Results](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/security-features/manifest-scan.gif)
+
+### During Helm App Deployment
+
+When you [deploy a helm chart](../user-guide/deploy-chart/deployment-of-charts.md), Devtron will scan the image associated with that helm chart and also the manifests, but unlike Devtron Apps, there is no code scan involved.
+
+Results of helm app scan will be visible under `Image Scan` and `Kubernetes Manifest` in the **App Details** page as shown below.
```
Comment (abhibhaw) [inline docs/user-guide/security-features.md:94]: GIF and steps needed for how to scan, a helm app

**Review by abhibhaw — CHANGES_REQUESTED** [review]
Missing scanning of kubernetes resources / cluster scanning etc. Entrypoints:
- Resource browser
- K8s resources on app details

File: docs/user-guide/security-features.md (hunk 2 above)
Comment (ashokdevtron) [reply docs/user-guide/security-features.md:26]: @abhibhaw, input for this data was from Nishant's webinar on DevSecOps. `When you commit the code, it's essential to scan it before building a container image` this has been written already. 

Can you please suggest what needs to be fixed here?

Comment (abhibhaw) [reply docs/user-guide/security-features.md:26]: ~After code commit~ Before building the artifact

File: docs/user-guide/security-features.md (hunk 3 above)
Comment (ashokdevtron) [reply docs/user-guide/security-features.md:94]: As per latest discussion, the **Configure** tab of Helm Apps is yet to have scanning options. Currently, it's scanned automatically; therefore, steps for end-user is not available.

File: docs/user-guide/security-features.md (hunk 1 above)
Comment (ashokdevtron) [reply docs/user-guide/security-features.md]: Fixed, removed the phrase altogether

File: docs/user-guide/security-features.md (hunk 2 above)
Comment (ashokdevtron) [reply docs/user-guide/security-features.md]: Changed the section title

**Review by abhibhaw — APPROVED** [review]
LGTM

**Review by uxarya-d — APPROVED** [review]

**Pull request merged**